modules/service/src/main/scala/ch/epfl/bluebrain/nexus/kg/service/queue/KafkaConsumer.scala

```diff
@@ -123,12 +145,10 @@ object KafkaConsumer {
     *
     * @param supervisor the supervisor actor handle
     */
-  @VisibleForTesting
   private[queue] def stop(supervisor: ActorRef)(implicit as: ActorSystem): Unit = {
     implicit val ec: ExecutionContext = as.dispatcher
     implicit val to: Timeout          = 30.seconds
-    val child                         = supervisor ? GetCurrentChild
-    child.foreach(_.asInstanceOf[CurrentChild].ref.get ! PoisonPill)
+    (supervisor ? GetCurrentChild).mapTo[KafkaConsumerWrapper[_]].foreach(_.kill())
```

Comment:
calling functions on the underlying actor breaks encapsulation, use messages instead even if this is only used in testing.